moveit_core/collision_distance_field/CMakeLists.txt

```diff
@@ -14,23 +9,53 @@ add_library(${MOVEIT_LIB_NAME}
   src/collision_world_hybrid.cpp
 )
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(${MOVEIT_LIB_NAME}
-  moveit_planning_scene moveit_distance_field
-  ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
 
-if(CATKIN_ENABLE_TESTING)
+ament_target_dependencies(${MOVEIT_LIB_NAME}
+  moveit_planning_scene
+  moveit_distance_field
+  urdf
+  visualization_msgs
+  tf2_eigen
+  angles
+)
+
+target_link_libraries(${MOVEIT_LIB_NAME}  
+  moveit_planning_scene
+  moveit_distance_field
+  moveit_collision_detection
+  moveit_robot_state
+  ${geometric_shapes_LIBRARIES}
+)
+  
+install(TARGETS ${MOVEIT_LIB_NAME}
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
+
+install(DIRECTORY include/ DESTINATION include)
+
+if(BUILD_TESTING)
+	find_package(ament_cmake_gtest REQUIRED)
   find_package(moveit_resources REQUIRED)
   include_directories(${moveit_resources_INCLUDE_DIRS})
+	find_package(resource_retriever REQUIRED)
 
-  catkin_add_gtest(test_collision_distance_field test/test_collision_distance_field.cpp)
-  target_link_libraries(test_collision_distance_field  ${MOVEIT_LIB_NAME})
-endif()
-
-install(TARGETS ${MOVEIT_LIB_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
+	if(WIN32)
+    # TODO add windows paths
+	else()
+		set(append_library_dirs "${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_BINARY_DIR}/../planning_scene;${CMAKE_CURRENT_BINARY_DIR}/../distance_field;${CMAKE_CURRENT_BINARY_DIR}/../collision_detection;${CMAKE_CURRENT_BINARY_DIR}/../robot_state;${CMAKE_CURRENT_BINARY_DIR}/../robot_model;${CMAKE_CURRENT_BINARY_DIR}/../utils")
```

Comment:
Using hardcoded paths is a bad idea. Is this really necessary?

Reply:
Hardcoding relative paths is a bad idea. 

I would recommend that for every library in this package that is used else ware you set a `<lib_name>_lib_dir` environment variable that can then be used by the tests

A good example is the `rcl_lib_dir` I have taken the time to show how it is used below:
```
/home/mike/ws_ros2/src/ros2/rcl/rcl/CMakeLists.txt:
   78  
   79: # rcl_lib_dir is passed as APPEND_LIBRARY_DIRS for each ament_add_gtest call so
   80  # the librcl that they link against is on the library path.
   81  # This is especially important on Windows.
   82  # This is overwritten each loop, but which one it points to doesn't really matter.
   83: set(rcl_lib_dir "$<TARGET_FILE_DIR:${PROJECT_NAME}>")
   84  

/home/mike/ws_ros2/src/ros2/rcl/rcl/test/CMakeLists.txt:
   16  
   17: set(extra_lib_dirs "${rcl_lib_dir}")
   18  
   
/home/mike/ws_ros2/src/ros2/rcl/rcl/test/CMakeLists.txt:
   34    rcl_add_custom_gtest(test_client${target_suffix}
   35      SRCS rcl/test_client.cpp
   36:     INCLUDE_DIRS ${osrf_testing_tools_cpp_INCLUDE_DIRS}
   37      ENV ${rmw_implementation_env_var}
   38      APPEND_LIBRARY_DIRS ${extra_lib_dirs}
```

Reply:
If you don't mind I prefer to merge this. And then I will fix them all together. Because we have other merged that use this style. I can open an issue to avoid forgetting it.